api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+
+/*The context class provides a context identifier*/
+class Context{
+  private:
+    /*The identifier itself*/
+    std::map<std::string, int> ctx_;
```

Comment:
Need to put this into consideration:
* According to the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/context.md#overview), "A Context MUST be immutable, and its write operations MUST result in the creation of a new Context containing the original values and the specified values updated".